source/mir/random/discrete.d

```diff
@@ -135,9 +137,10 @@ struct Discrete(T)
             arr[sm.index] = AltPair(sm.prob, lg.index);
 
             // update larger & reinsert
-            lg.prob += sm.prob - 1;
+            lg.prob += sm.prob;
+            lg.prob -= 1;
 
-            if (lg.prob < 1)
+            if (lg.prob < 1 || lg.prob.approxEqual(1, 1e-13, 1e-13))
```

Comment:
> No need. This is ad-hoc for Tinflex testing and it should not be a part of discrete

Well I just thought it's nice that if one writes tests with a seeded test for his algorithm the "randomness" should go away, e.g. sth like the following could yield _different_ results depending on 32/64.

```
auto ds = Discrete!double(0.25, 0.5, 0.25);
auto gen = Mt19937(42);
ds(gen);
```

Should we maybe add it as a template argument?